crates/optimism/txpool/src/maintain.rs

```diff
@@ -63,10 +77,52 @@ where
                 timestamp: new.tip().timestamp(),
             };
             let mut to_remove = Vec::new();
+            let mut to_revalidate = Vec::new();
             for tx in &pool.pooled_transactions() {
                 if tx.transaction.has_exceeded_block_attributes(&block_attr) {
                     to_remove.push(*tx.hash());
                 }
+                // Only interop txs have this field set
+                if let Some(interop) = tx.transaction.interop() {
+                    if !interop.is_valid(block_attr.timestamp) {
+                        // That means tx didn't revalidated during [`OFFSET_TIME`] time
+                        // We could assume that it won't be validated at all and remove it
+                        to_remove.push(*tx.hash());
+                    } else if interop.is_stale(block_attr.timestamp, OFFSET_TIME) {
+                        // If tx has less then [`OFFSET_TIME`] of valid time we revalidate it
+                        to_revalidate.push(tx.clone())
+                    }
+                }
+            }
+            if !to_revalidate.is_empty() {
+                // TODO: We should change it to a single batch call once supervisor implements it
+                // TODO: add metrics for revalidation
+                for tx in to_revalidate {
+                    if let Some(Err(err)) = is_valid_cross_tx(
+                        tx.transaction.access_list(),
+                        tx.transaction.hash(),
+                        block_attr.timestamp,
+                        Some(TRANSACTION_VALIDITY_WINDOW),
+                        // We could assume that interop is enabled, because
+                        // tx.transaction.interop() would be set only in
+                        // this case
+                        true,
+                        Some(&supervisor_client),
+                    )
+                    .await
```

Comment:
these requests we should join or even buffer to prevent an unbounded # of cross chain events at once.

this feature can be added directly into the Supervisorclient

like `check_many` that accepts a list of tx hashes instead and returns a Vec of `result<txhash, err,txhash>`

that then creates a vec of requests and makes use of

https://docs.rs/futures/latest/futures/stream/trait.StreamExt.html#method.buffered with some sane concurrency factor, like 10 for 10 requests at once


later this could also be replaced with batch requests

Reply:
Actually I wanted to join_all but forgot :)

Reply:
Done! (got my arse kicked by async a bit) 
i expect supervisor to add special method to validate multiple txs, so i'm up to keeping the logic in the maintain function, and later we would remove all futures and just pass it in supervisor